[catch2] build failure

<!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
<!-- ⁉ You can find its location at the end of the console output. ⁉ -->

**Host Environment**

- OS: [e.g. Windows/Linux etc...] Windows 11 [Version 10.0.22621.675]
- Compiler: Visual Studio Community 2022 Version: [17.3.6]  VS Tools Version: [14.32.31326] 
Microsoft (R) C/C++ Optimizing Compiler Version [19.32.31335] for x64 

**To Reproduce**

Steps to reproduce the behavior:

`./vcpkg install catch2`

**Failure logs**

- Cut and paste the appropriate build messages from the console output.
- -- Note: catch2 only supports static library linkage. Building static library.
-- Downloading https://github.com/catchorg/Catch2/archive/v3.1.1.tar.gz -> catchorg-Catch2-v3.1.1.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Feature flag 'binarycaching' = off
[DEBUG] Feature flag 'manifests' = off
[DEBUG] Feature flag 'compilertracking' = off
[DEBUG] Feature flag 'registries' = off
[DEBUG] Feature flag 'versions' = off
[DEBUG] Detected CI environment: VCPKG_NO_CI
[DEBUG] Downloading https://github.com/catchorg/Catch2/archive/v3.1.1.tar.gz
Error: Failed to download from mirror set:
File does not have the expected hash:
             url : [ https://github.com/catchorg/Catch2/archive/v3.1.1.tar.gz ]
       File path : [ D:\vcpkg-downloads\catchorg-Catch2-v3.1.1.tar.gz.22096.part ]
   Expected hash : [ d056c8bad8407364debff9f626c7f8255eb64e0faf6dc7fba3d7e14ed29a7e59bd45d92d7b3819bb67d6299a2024f5949c755995d98e94c9bb3d2013709e57ef ]
     Actual hash : [ b6deae2e87ae301bb3de9e378297a3f20333bb5f233fd9ef81f8c4f10fe0c4ad5de9174b850da8064607b9c56c68c7a0815e2bac80108e6db905bc660def7d57 ]


[DEBUG] D:\vcpkg-tool\src\vcpkg\base\downloads.cpp(766): 
[DEBUG] Time in subprocesses: 0 us
[DEBUG] Time in parsing JSON: 9 us
[DEBUG] Time in JSON reader: 0 us
[DEBUG] Time in filesystem: 54 us
[DEBUG] Time in loading ports: 0 us
[DEBUG] Exiting after 989.8 ms (989736 us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
      
      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.
      
      3. Your proxy's remote server is out of service.
      
      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
      and (https://github.com/microsoft/vcpkg-tool/pull/77)
      
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:273 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/catch2/portfile.cmake:5 (vcpkg_from_github)
  scripts/ports.cmake:147 (include)

- Attach ***every failure log file*** mentioned in the console output.
[stdout-x64-windows.log](https://github.com/microsoft/vcpkg/files/9823906/stdout-x64-windows.log)
[x64-windows.vcpkg_abi_info.txt](https://github.com/microsoft/vcpkg/files/9823907/x64-windows.vcpkg_abi_info.txt)

**Additional context**
It seems the archive was changed for version 3.1.1 (on 20221019 Local) after the PR [catch2] Update to 3.1.1 #27282 was committed (on 20221018 Local), thus the archive sha512 hash has changed. 

Add any other context about the problem here, such as what you have already tried to resolve the issue.
* PR PR [catch2] Update to 3.1.1 #27282
![image](https://user-images.githubusercontent.com/68293825/196794742-7d374cf7-3be5-4203-b673-330d79ee3995.png)


* https://github.com/catchorg/Catch2/releases/tag/v3.1.1
![image](https://user-images.githubusercontent.com/68293825/196794872-71e17153-945d-4dc7-ab2b-8acac474516a.png)

--- a/ports/catch2/portfile.cmake
+++ b/ports/catch2/portfile.cmake
@@ -6,7 +6,7 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO catchorg/Catch2
     REF v3.1.1
-    SHA512 d056c8bad8407364debff9f626c7f8255eb64e0faf6dc7fba3d7e14ed29a7e59bd45d92d7b3819bb67d6299a2024f5949c755995d98e94c9bb3d2013709e57ef
+    SHA512 b6deae2e87ae301bb3de9e378297a3f20333bb5f233fd9ef81f8c4f10fe0c4ad5de9174b850da8064607b9c56c68c7a0815e2bac80108e6db905bc660def7d57
     HEAD_REF devel
     PATCHES
         fix-install-path.patch

--- a/ports/catch2/vcpkg.json
+++ b/ports/catch2/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "catch2",
   "version-semver": "3.1.1",
+  "port-version": 1,
   "description": "A modern, header-only test framework for unit testing.",
   "homepage": "https://github.com/catchorg/Catch2",
   "license": "BSL-1.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1314,7 +1314,7 @@
     },
     "catch2": {
       "baseline": "3.1.1",
-      "port-version": 0
+      "port-version": 1
     },
     "cccapstone": {
       "baseline": "9b4128ee1153e78288a1b5433e2c06a0d47a4c4e",

--- a/versions/c-/catch2.json
+++ b/versions/c-/catch2.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d83c493274ecc1798d1d4ac05bf83eedec161695",
+      "version-semver": "3.1.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "44a94453e68878bf44df00e51bae5115dff3baa8",
       "version-semver": "3.1.1",
       "port-version": 0

